### PR TITLE
fix: crash when selecting some categories from skills activity

### LINKED ISSUE #2115: App crash on selecting some specific categories from Categories section
**Describe the bug**
App crashes when selecting categories like Movies and TV, Weather, etc.

**Expected Behaviour**
The app should show corresponding skills instead of crashing.

**Steps to reproduce it**
Select any category like Movies and TV, Weather, etc from skills activity.

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by naman653**
Fixes #2115 #1705 

Reasons of Crash: 
Skills are fetched in form of SkillData which defines most of its attributes as non-nullable. But many skills coming from the server stores those attributes values as null which causes Exceptions when different methods are called on them. These exceptions inclued IllegalArgumentException, NullPointerException, IndexOutOfBoundException, etc. This causes the app to crash.

Changes: I've handled the exceptions to avoid the crash.

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt [hunk 1]
```diff
@@ -33,38 +34,42 @@ class SkillsListAdapter(
...
+                        .error(R.drawable.ic_susi)
+                        .transform(CircleTransform())
+                        .into(holder.skillImage)
+            }
+            val stars = skillData.skillRating?.stars
+            if (stars != null) {
+                holder.skillRating.rating = stars.averageStar
+                holder.skillTotalRatings.text = stars.totalStar.toString()
+            }
+        } catch (exception: Exception) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt]: Never catch generic exception

Comment (naman653) [reply app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt]: I know, but there are a lot of exceptions happening, so I did that. Well, I will improve that and try to catch each and every exception separately.

File: app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt [hunk 2]
```diff
@@ -33,38 +36,44 @@ class SkillsListAdapter(
...
+                        .into(holder.skillImage)
+            }
+            val stars = skillData.skillRating?.stars
+            if (stars != null) {
+                holder.skillRating.rating = stars.averageStar
+                holder.skillTotalRatings.text = stars.totalStar.toString()
+            }
+        } catch (illegalArgument: IllegalArgumentException) {
+            Timber.e(illegalArgument)
+        } catch (nullPointer: NullPointerException) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt]: Don't catch NPE

File: app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt [hunk 3]
```diff
@@ -33,38 +36,44 @@ class SkillsListAdapter(
...
+                        .error(R.drawable.ic_susi)
+                        .transform(CircleTransform())
+                        .into(holder.skillImage)
+            }
+            val stars = skillData.skillRating?.stars
+            if (stars != null) {
+                holder.skillRating.rating = stars.averageStar
+                holder.skillTotalRatings.text = stars.totalStar.toString()
+            }
+        } catch (illegalArgument: IllegalArgumentException) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt]: Why is IllegalArgumentException thrown?

File: app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt (hunk 2 above)
Comment (naman653) [reply app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt]: **Logcat of the Crash**
java.lang.NullPointerException: Attempt to invoke interface method 'int java.lang.CharSequence.length()' on a null object reference
        at org.fossasia.susi.ai.skills.groupwiseskills.adapters.recycleradapters.SkillsListAdapter.onBindViewHolder(SkillsListAdapter.kt:39)
        at org.fossasia.susi.ai.skills.groupwiseskills.adapters.recycleradapters.SkillsListAdapter.onBindViewHolder(SkillsListAdapter.kt:23)
        at android.support.v7.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:6781)
        at android.support.v7.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:6823)
        at android.support.v7.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:5752)
        at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6019)
        at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5858)
        at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5854)
        at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2230)
        at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1557)
        at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1517)
        at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:612)
        at android.support.v7.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:3924)
        at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:3641)
        at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:4194)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:639)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:574)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.support.v4.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:625)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:639)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:574)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:639)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:574)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.support.v7.widget.ActionBarOverlayLayout.onLayout(ActionBarOverlayLayout.java:444)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:639)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:574)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1959)
        at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1813)
        at android.widget.LinearLayout.onLayout(LinearLayout.java:1722)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.widget.FrameLayout.layoutChildren(FrameLayout.java:639)
        at android.widget.FrameLayout.onLayout(FrameLayout.java:574)
        at android.view.View.layout(View.java:16001)
        at android.view.ViewGroup.layout(ViewGroup.java:5181)
        at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2484)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2181)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1293)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6599)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:800)
        at android.view.Choreographer.doCallbacks(Choreographer.java:603)
    	at android.view.Choreographe

File: app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt (hunk 3 above)
Comment (naman653) [reply app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt]: I was getting an error before due to this, but I think it's working fine without it. So no need for this.

File: app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt (hunk 2 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/adapters/recycleradapters/SkillsListAdapter.kt]: Yeah, so add null check

**Pull request merged**